### ISSUE TITLE
[Bug] How to use Dark Mode?

### ISSUE THREAD

**Issue by garv-shah** [AppFlowy-IO/appflowy-editor#258]
### Bug Description

Using `editorStyle: EditorStyle.dark` with `AppFlowyEditor` does not seem to work, in that it does not change any theming elements. Similarly, changing any property on `editorStyle` does not seem to change the theme of the editor.
When running the example on. the main branch, it seems like "Built In Dark Mode" and "Custom Theme" were intended features, but neither of the buttons seem to do anything.
How would I be able to implement dark mode in my app? I've tried consulting the documentation but it seems to be a but outdated. Thanks!

### How to Reproduce

1. Clone the main branch
2. Run the example app
3. Open the app drawer and click either of the Theme Demos (in my experience nothing happened)

Alternatively:

1. Set up an example `AppFlowyEditor` as shown in the README
2. Change any attribute on `editorStyle`
3. Observe that no change has been made to the editor

### Expected Behavior

I expected for the editor to automatically adapt to the current Flutter theme, or for manual theming to have an impact on the style of the editor.

For example, the following code should produce a dark mode editor, but it does not seem to.

```dart
AppFlowyEditor.standard(
    editorState: editorState,
    editorStyle: EditorStyle.dark
)
```

### Operating System

MacOS Ventura and iOS 16

### AppFlowy Editor Version(s)

main branch

### Screenshots

<img width="944" alt="image" src="https://github.com/AppFlowy-IO/appflowy-editor/assets/64292655/0c825c6f-88c6-4ebb-8edc-1f6470f3aaff">

**Comment by garv-shah** [comment]
I got it working by using the `textStyleConfiguration` property as follows, where `Theme.of(context).primaryColorLight` can be any colour you'd like to text to be

```dart
editorStyle: const EditorStyle.desktop().copyWith(
  textStyleConfiguration: TextStyleConfiguration(
    text: TextStyle(
      color: Theme.of(context).primaryColorLight,
    )
  )
)
```

Still, the `textStyle` property does not work, which is what `EditorStyle.dark` uses, so it should either be depreciated, or updated to properly reflect the new theme

**Comment by LucasXu0** [comment]
Hey, @garv-shah. The `EditorStyle.dark` is deprecated. You can customize the editor style whatever you want by using `TextStyleConfiguration`.

**Comment by LucasXu0** [comment]
For example, this is how I customize the editor style in AppFlowy.

```dart
  EditorStyle desktop() {
    final theme = Theme.of(context);
    final fontSize = context.read<DocumentAppearanceCubit>().state.fontSize;
    return EditorStyle.desktop(
      padding: padding,
      backgroundColor: theme.colorScheme.surface,
      cursorColor: theme.colorScheme.primary,
      textStyleConfiguration: TextStyleConfiguration(
        text: TextStyle(
          fontFamily: 'Poppins',
          fontSize: fontSize,
          color: theme.colorScheme.onBackground,
          height: 1.5,
        ),
        bold: const TextStyle(
          fontFamily: 'Poppins-Bold',
          fontWeight: FontWeight.w600,
        ),
        italic: const TextStyle(fontStyle: FontStyle.italic),
        underline: const TextStyle(decoration: TextDecoration.underline),
        strikethrough: const TextStyle(decoration: TextDecoration.lineThrough),
        href: TextStyle(
          color: theme.colorScheme.primary,
          decoration: TextDecoration.underline,
        ),
        code: GoogleFonts.robotoMono(
          textStyle: TextStyle(
            fontSize: fontSize,
            fontWeight: FontWeight.normal,
            color: Colors.red,
            backgroundColor: theme.colorScheme.inverseSurface,
          ),
        ),
      ),
    );
```

**Comment by garv-shah** [comment]
oh nice perfect, thank you for that code snippet it worked perfectly!
I'll try to submit a quick PR updating the documentation, since it was a bit hard to navigate

**Comment by garv-shah** [comment]
Hi @annieappflowy, sure I'd love to!

I'm using `appflowy_editor` for an education app platform which encompasses a tutoring app, an app for my school's maths club and an app for orphanages as of now. Namely, I'm using the editor to streamline the process of making questions, quizzes and posts within the app, so it is easy to curate content.

I recently switched to the package because the other WYSIWYG Flutter packages were quite painful to use. I used [Flutter Quill](https://github.com/singerdmx/flutter-quill) initially, but the code quality of that project is quite hard to navigate, and since it is so frequently added to, I'd find myself dealing with breaking changes every time I wanted to work on my project rather than implementing the features I want. 
This issue with stability landed me on [Visual Editor](https://github.com/visual-space/visual-editor), a fork of Flutter Quill that aimed to address the many issues I had with the parent package. It worked quite well for about a year, until a few months ago where I noticed many detrimental bugs were arising. These include the backspace key or arrow navigation not working on MacOS, the package causing the app to fail compiling on iOS and a couple visual glitches. It has been a few months since these were introduced, so a couple days ago I tried switching over to another editor, which has landed me on App Flowy.

Another reason that App Flowy was appealing is due to the plugin system. Being education, my apps need to support maths formulas, meaning that TeX support was something I had to implement myself in both the other packages. The actual AppFlowy app already has this functionality built-in thanks to the maths plugin, which was ideal for my use case.

The only small things that I hope will be or continue to be supported are as follows:
1. Better Quill Delta support. My previous database was all in Quill Delta, so more import export features would be wonderful. This could include pdf exports too!
2. Up to date documentation. The code quality for the editor is already wonderful, and it was very easy to implement features I wanted to, but the documentation sometimes did not indicate the depreciation of certain parameters and functions.
3. Customisability! The more custom use cases that the editor can be used for the better, and the AppFlowy app is a great example to take inspiration from. It would be great if the editor does not stay hard-coded but easily customisable.
4. Ongoing stability. I've very much enjoyed using the package over the last few days, and I'm hoping it will stay stable and up to date on the latest version of Flutter going forward :)

Thank you so much for all the work you guys have put into this package, it's a joy to use and has already improved my apps significantly. Thanks for reading my feedback :D
